src/components/Tooltip/Tooltip.tsx

```diff
@@ -56,113 +59,263 @@ const paddingsBottom = {
+            }
+        },
+        [triggerRefElement?.current, tooltipContainerRef?.current],
+    );
+
+    const hasInteractiveElements = !!(buttons?.length || linkUrl?.length);
 
     return (
-        <>
+        <div>
             <div
-                ref={tooltipContainerRef}
-                className="tw-popper-container tw-inline-block tw-max-w-[200px] tw-bg-black-100 dark:tw-bg-white tw-rounded-md tw-shadow-mid tw-text-white dark:tw-text-black-100 tw-z-20"
-                style={style}
-                data-test-id="tooltip"
-                {...popperAttributes}
-                {...tooltipProps}
+                onMouseOver={(event) => checkIfHovered(event.nativeEvent)}
+                onMouseLeave={handleHideTooltipOnHover}
+                onFocus={() => setIsOpen(true)}
+                onBlur={() => (!hasInteractiveElements ? setIsOpen(false) : null)}
             >
-                {brightHeader && <BrightHeader headerStyle={brightHeader} />}
-                <div
-                    className={merge([
-                        "tw-px-4",
-                        hasLargePaddingTop ? paddingsTop.small : paddingsTop.large,
-                        linkUrl ? paddingsBottom.small : paddingsBottom.large,
-                    ])}
-                >
-                    {heading && (
-                        <h4 className="tw-flex tw-text-m tw-font-bold tw-mb-1">
-                            {headingIcon && (
-                                <span className="tw-mr-1.5">
-                                    {cloneElement(headingIcon, { size: IconSize.Size20 })}
-                                </span>
-                            )}
-                            {heading}
-                        </h4>
-                    )}
-                    <div className="tw-flex">
-                        {tooltipIcon && (
-                            <span className="tw-shrink-0 tw-mr-1">
-                                {cloneElement(tooltipIcon, { size: IconSize.Size16 })}
-                            </span>
-                        )}
-                        <p className="tw-text-s">{content}</p>
-                    </div>
-                    {linkUrl && (
-                        <a
-                            {...mergeProps(linkProps, focusProps)}
-                            data-test-id="tooltip-link"
-                            ref={linkRef}
-                            href={linkUrl}
-                            target="_blank"
-                            rel="noopener noreferrer"
-                            className={merge([
-                                "tw-text-xs tw-text-black-40 dark:tw-text-black-80 tw-underline tw-mt-1",
-                                isFocusVisible && FOCUS_STYLE,
-                            ])}
+                {triggerElement &&
+                    cloneElement(triggerElement, {
+                        ref: triggerRefElement,
+                    })}
+            </div>
```

Comment:
Wrapping the trigger element in a div brings about the same issues that we had with the FlyoutTrigger, where you might have a component that directly relies on it's parent for styling and cannot anymore. I would maybe suggest a render function.
`const triggerProps = {onMouseOver, onMouseLeave, onFocus, onBlur}`
```
<>
  {triggerElement(triggerProps, triggerRef)}
</>
```
wdyt?

Reply:
Will create a task for it.